CNKI.js

```diff
@@ -38,47 +38,50 @@
 
 // Fetches RefWorks records for provided IDs and calls onDataAvailable with resulting text
 // ids should be in the form [{dbname: "CDFDLAST2013", filename: "1013102302.nh"}]
-function getRefWorksByID(ids, onDataAvailable) {
-	if (!ids.length) return;
-	var { dbname, filename, url } = ids.shift();
-	let postData = "filename=" + filename + 
-		"&displaymode=Refworks&orderparam=0&ordertype=desc&selectfield=&dbname=" + 
-		dbname + "&random=0.2111567532240084";
-	
-	ZU.doPost('https://kns.cnki.net/KNS8/manage/ShowExport', postData,
-		function (text) {
-			let data = text
-				.replace("<ul class='literature-list'><li>", "")
-            	.replace("<br></li></ul>", "")
-            	.replace("</li><li>", "") // divide results
-            	.replace(/<br>|\r/g, "\n")
-            	.replace(/vo (\d+)\n/, "VO $1\n") // Divide VO and IS to different line
-            	.replace(/IS 0(\d+)\n/g, "IS $1\n")  // Remove leading 0
-            	.replace(/VO 0(\d+)\n/g, "VO $1\n")
-            	.replace(/\n+/g, "\n")
-            	.replace(/\n([A-Z][A-Z1-9]\s)/g, "<br>$1")
-            	.replace(/\n/g, "")
-            	.replace(/<br>/g, "\n")
-            	.replace(/\t/g, "") // \t in abstract
-            	.replace(
-            	    /^RT\s+Conference Proceeding/gim,
-            	    "RT Conference Proceedings"
-            	)
-            	.replace(/^RT\s+Dissertation\/Thesis/gim, "RT Dissertation")
-            	.replace(/^(A[1-4]|U2)\s*([^\r\n]+)/gm, function (m, tag, authors) {
-            	    authors = authors.split(/\s*[;，,]\s*/); // that's a special comma
-            	    if (!authors[authors.length - 1].trim()) authors.pop();
-            	    return tag + " " + authors.join("\n" + tag + " ");
-            	})
-            	.trim();
-			// Z.debug(data);
-			onDataAvailable(data, url);
-			// If more results, keep going
-			if (ids.length) {
-				getRefWorksByID(ids, onDataAvailable);
+async function getRefWorksByID(id) {
+	if (!id) return;
+	var { dbname, filename, url } = id;
+	const postData = `FileName=${dbname}!${filename}!1!0&DisplayMode=Refworks&OrderParam=0&OrderType=desc&SelectField=&PageIndex=1&PageSize=20&language=&uniplatform=NZKPT&random=0.30585230060685187`;
+	const refer = `https://kns.cnki.net/dm/manage/export.html?filename=${dbname}!${filename}!1!0&displaymode=NEW&uniplatform=NZKPT`;
+	let reftext = await request(
+		'https://kns.cnki.net/dm/api/ShowExport',
+		{
+			method: "POST",
+			body: postData,
+			responseType: "application/x-www-form-urlencoded",
```

Comment:
This isn't a [valid value for an XHR response](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType). What are you trying to do here?

Reply:
Sorry for that. I added this code because it just works. I tested the API in tools like Postman.  I used to set `Content-Type: application/x-www-form-urlencoded`, but it did not work. So I searched the Code in Zotero translator to find some solutions. I tried this `responseType: "application/x-www-form-urlencoded"` and it works.

But I update the code, the below is more reasonable
```js
let reftext = await request(
		'https://kns.cnki.net/dm/api/ShowExport',
		{
			method: "POST",
			body: postData,
			headers: {
				Referer: refer
			}
		}
	);
``

Reply:
`responseType` doesn't have anything to do with a `Content-Type` request header, though. `application/x-www-form-urlencoded` is just invalid for `responseType`. That's not used in any translators and definitely wasn't doing anything for you.